icu4c/source/i18n/udateintervalformat.cpp

```diff
@@ -128,8 +128,27 @@ udtitvfmt_formatToResult(
     }
     auto* resultImpl = UFormattedDateIntervalApiHelper::validate(result, *status);
     DateInterval interval = DateInterval(fromDate,toDate);
-    resultImpl->fImpl = reinterpret_cast<const DateIntervalFormat*>(formatter)
-        ->formatToValue(interval, *status);
+    if (resultImpl != nullptr) {
+        resultImpl->fImpl = reinterpret_cast<const DateIntervalFormat*>(formatter)
+            ->formatToValue(interval, *status);
+    }
+}
+
+U_DRAFT void U_EXPORT2
+udtitvfmt_formatCalendarToResult(
+                const UDateIntervalFormat* formatter,
+                UCalendar*      fromCalendar,
+                UCalendar*      toCalendar,
+                UFormattedDateInterval* result,
+                UErrorCode*     status) {
+    if (U_FAILURE(*status)) {
+        return;
+    }
+    auto* resultImpl = UFormattedDateIntervalApiHelper::validate(result, *status);
+    if (resultImpl != nullptr) {
+        resultImpl->fImpl = reinterpret_cast<const DateIntervalFormat*>(formatter)
+            ->formatToValue(*(Calendar *)fromCalendar, *(Calendar *)toCalendar, *status);
```

Comment:
Although unpleasant, this is consistent with the pattern used for UCalendar in ucal.cpp (C-style cast, no type check or null check)

Reply:
> Although unpleasant, this is consistent with the pattern used for UCalendar in ucal.cpp (C-style cast, no type check or null check)

Notice this is different from ucal.cpp. In ucal.cpp, it is the direct use of the passing parameter after casting. But in here, the pointer is not directly passing in from the coder, but return from the validate function. There are no guarantee that while the input parameter is not null the validate will always return not null.

Reply:
The use of `result` is guarded, but `fromCalendar` and `toCalendar` are not guarded.  But again, that's consistent with ucal.cpp, so no action required, but if you wanted to be extra safe you could do a null or branding check.